android/java/org/chromium/chrome/browser/crypto_wallet/activities/AccountDetailActivity.java

```diff
@@ -173,40 +184,43 @@ private void setUpAssetList(NetworkInfo selectedNetwork) {
 
     private void fetchAccountInfo(NetworkInfo selectedNetwork) {
         assert mKeyringService != null;
-        mKeyringService.getKeyringInfo(AssetUtils.getKeyringForCoinType(mCoinType), keyringInfo -> {
-            if (keyringInfo == null || mWalletModel == null) {
-                return;
-            }
-
-            AccountInfo[] accounts = keyringInfo.accountInfos;
-            LiveDataUtil.observeOnce(
-                    mWalletModel.getCryptoModel().getNetworkModel().mCryptoNetworks,
-                    allNetworks -> {
-                        Utils.getTxExtraInfo(new WeakReference<>(this), TokenUtils.TokenType.ALL,
-                                allNetworks, selectedNetwork, accounts, null, false,
-                                (assetPrices, fullTokenList, nativeAssetsBalances,
-                                        blockchainTokensBalances) -> {
-                                    for (AccountInfo accountInfo : accounts) {
-                                        if (accountInfo.address.equals(mAddress)
-                                                && accountInfo.name.equals(mName)) {
-                                            AccountInfo[] accountInfos = new AccountInfo[1];
-                                            accountInfos[0] = accountInfo;
-                                            WalletListItemModel thisAccountItemModel =
-                                                    new WalletListItemModel(
-                                                            Utils.getCoinIcon(mCoinType), mName,
-                                                            mAddress, null, null, mIsImported);
-                                            Utils.setUpTransactionList(this, accountInfos,
-                                                    thisAccountItemModel, assetPrices,
-                                                    fullTokenList, nativeAssetsBalances,
-                                                    blockchainTokensBalances,
-                                                    findViewById(R.id.rv_transactions), this,
-                                                    mWalletTxCoinAdapter);
-                                            break;
-                                        }
-                                    }
-                                });
-                    });
-        });
+        mKeyringService.getKeyringInfo(
+                AssetUtils.getKeyringForChainId(selectedNetwork.chainId), keyringInfo -> {
+                    if (keyringInfo == null || mWalletModel == null) {
+                        return;
+                    }
+
+                    AccountInfo[] accounts = keyringInfo.accountInfos;
+                    LiveDataUtil.observeOnce(
+                            mWalletModel.getCryptoModel().getNetworkModel().mCryptoNetworks,
+                            allNetworks -> {
+                                Utils.getTxExtraInfo(new WeakReference<>(this),
```

Comment:
Wondering what is the point to pass weak reference here since callback lambda will capture "this" anyway?

Reply:
Could we spit into several methods in this PR or log an issue for refactoring since call stack is too deep and it is hard to read it?

Reply:
Yes, you're right! We should keep passing a weak reference if we want to avoid memory leaks, and this part is also cumbersome to read. I prefer to file a new GitHub issue as the refactoring may end up being slightly more time consuming here.

Reply:
Filed new GitHub issue 👉 https://github.com/brave/brave-browser/issues/30135